source/l10nUtil.py

```diff
@@ -572,12 +572,15 @@ def check(self) -> bool:
 			return False
 		return True
 
-	# e.g. %s %d (but not %%)
-	RE_UNNAMED_PERCENT = re.compile(r"(?<!%)%(?!%)[.\d]*[a-zA-Z]")
+	# e.g. %s %d %10.2f %-5s (but not %%) or %%(name)s %(name)d
+	RE_UNNAMED_PERCENT = re.compile(
+		# This regex is magic generated by AI. Ensure it is thoroughly unit tested.
+		r"(?<!%)%(?!%)[-#+0]*(?:\*|\d+)?(?:\.(?:\*|\d+))?[hlL]?[a-zA-Z]|(?<=%%)%(?!%)[-#+0]*(?:\*|\d+)?(?:\.(?:\*|\d+))?[hlL]?[a-zA-Z]",
+	)
```

Comment:
I'd still suggest thorough unit testing, but this seems to pass all the tests, and was hand-written by yours truly by reading the [printf style formatting documentation](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting).
```suggestion
	RE_UNNAMED_PERCENT = re.compile(
		# Does not include optional mapping key, as that's handled by a different regex
		r"""
		(?:(?<=%%)|(?<!%))?%  # Per cent sign, optionally preceded by 2 percent signs, but not by just 1
		[#0+-]*  # Optional conversion flags
		\d*  # Optional minimum field width
		(?:\.\d+)?  # Optional precision specifier - if present, must be a full stop followed by 1-or-more digits
		[hlL]?  # Optional length specifier - has no effect in Python
		[diouxXeEfFgGcrsa]  # Conversion type
		""",
		flags=re.VERBOSE
	)
```